common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,166 @@
+
+    @Override
+    public void onEnable() {
+        Pl3xMap.api().getEventRegistry().register(new Pl3xEvents(this));
+
+        if (Pl3xMap.api().isEnabled()) {
+            Pl3xMap.api().getWorldRegistry().forEach(this::registerLayers);
+        }
+
+        plugin.log(Level.INFO, "Enabled Pl3xMap markers hook. Populating web map with claims...");
+        for (World world : plugin.getWorlds()) {
+            plugin.getClaimWorld(world).ifPresent(claimWorld -> setClaimMarkers(claimWorld.getClaims(plugin), world));
+        }
+    }
+
+    @Override
+    public void setClaimMarker(@NotNull TownClaim claim, @NotNull World world) {
+        plugin.getClaimWorld(world).ifPresent(claimWorld -> {
+            ConcurrentLinkedQueue<TownClaim> townClaims = claims.computeIfAbsent(claimWorld, k -> new ConcurrentLinkedQueue<>());
+            townClaims.add(claim);
+        });
+    }
+
+    @Override
+    public void removeClaimMarker(@NotNull TownClaim claim, @NotNull World world) {
+        plugin.getClaimWorld(world).ifPresent(claimWorld -> {
+            ConcurrentLinkedQueue<TownClaim> townClaims = claims.get(claimWorld);
+            townClaims.remove(claim);
+        });
+    }
+
+    @Override
+    public void setClaimMarkers(@NotNull List<TownClaim> claims, @NotNull World world) {
+        claims.forEach(claim -> setClaimMarker(claim, world));
+    }
+
+    @Override
+    public void removeClaimMarkers(@NotNull List<TownClaim> claims, @NotNull World world) {
+        claims.forEach(claim -> removeClaimMarker(claim, world));
+    }
+
+    @Override
+    public void clearAllMarkers() {
+        claims.clear();
+    }
+
+    @NotNull
+    private String getLayerName() {
+        return plugin.getSettings().getWebMapMarkerSetName();
+    }
+
+    @NotNull
+    private String getClaimMarkerKey(@NotNull TownClaim claim, @NotNull net.pl3x.map.core.world.World world) {
+        return plugin.getKey(
+                claim.town().getName().toLowerCase(),
+                Integer.toString(claim.claim().getChunk().getX()),
+                Integer.toString(claim.claim().getChunk().getZ()),
+                world.getName()
+        ).toString();
+    }
+
+    private void registerLayers(@NotNull net.pl3x.map.core.world.World mapWorld) {
+        if (plugin.getSettings().doWebMapHook()) {
+            ClaimsLayer layer = new ClaimsLayer(this, mapWorld);
+            mapWorld.getLayerRegistry().register(layer);
+        }
+    }
+
+    @NotNull
+    public Options getMarkerOptions(@NotNull TownClaim claim) {
+        final TextColor color = claim.town().getDisplayColor();
+        return Options.builder()
+                .tooltip(new Tooltip(claim.town().getName()).setDirection(Tooltip.Direction.TOP))
+                .fillColor(Colors.argb(255 / 2, color.red(), color.green(), color.blue()))
+                .strokeColor(Colors.rgb(Math.max((int) (color.red() * 0.7), 0), Math.max((int) (color.green() * 0.7), 0), Math.max((int) (color.blue() * 0.7), 0)))
```

Comment:
Is there ever a state where `color.red()/blue()/green()` could return a negative value and thus trigger the maximization logic to select 0? Those methods should always return an int in the range of 0-255, so the lowest they could be is `0` (and `0 * 0.7 = 0` still)

Reply:
To be honest I just ripped this code from the awt darker method, not too sure why it's a thing but just trusted the process haha

Reply:
Resolved in [de230d8](https://github.com/WiIIiam278/HuskTowns/pull/303/commits/de230d873d6c7bd4f1f4fdff86a63ac5b9f99ffc)